pkg/providers/aws.go

```diff
@@ -17,12 +22,50 @@ import (
 
 	"github.com/openshift/cluster-api-actuator-pkg/pkg/framework"
 	"github.com/openshift/cluster-api-actuator-pkg/pkg/framework/gatherer"
+	"github.com/tidwall/gjson"
 )
 
 const (
 	amiIDMetadataEndpoint = "http://169.254.169.254/latest/meta-data/ami-id"
 )
 
+// createAWSClient create AWS client.
+func createAWSClient(oc *gatherer.CLI) *framework.AwsClient {
+	awscreds, err := oc.WithoutNamespace().Run("get").Args("secret/aws-creds", "-n", "kube-system", "-o", "json").Output()
+	if err != nil {
+		Skip("Unable to get AWS credentials secret, skipping the testing.")
+	}
+
+	accessKeyIDBase64, secureKeyBase64 := gjson.Get(awscreds, `data.aws_access_key_id`).String(), gjson.Get(awscreds, `data.aws_secret_access_key`).String()
+
+	accessKeyID, err1 := base64.StdEncoding.DecodeString(accessKeyIDBase64)
```

Comment:
non blocking nit here: Generally, redefining the error `err1`, `err2` etc can make sense to ensure that we don't have variable shadowing, in this instance I'm not sure it's helping us. (It's not harming us either, hence non blocking nit)

Beause the steps are executed in sequence, if we return an error  at any point, the following `Expect(errN).NotTo(HaveOccurred())` will fail. This means we could define a single `err` and reuse it without worring about shadowing. 


```suggestion
	accessKeyID, err := base64.StdEncoding.DecodeString(accessKeyIDBase64)
```